spring/src/main/java/org/axonframework/spring/messaging/DefaultEventMessageConverter.java

```diff
@@ -102,7 +102,7 @@ public <T> EventMessage<T> convertFromInboundMessage(Message<T> message) {
     private static <T> QualifiedName getName(Message<T> message) {
         MessageHeaders headers = message.getHeaders();
         return headers.containsKey(MESSAGE_NAME)
-                ? QualifiedName.fromString(Objects.toString(headers.get(MESSAGE_NAME)))
+                ? QualifiedName.fromString(Objects.toString(headers.get(MESSAGE_NAME))) // todo: QualifiedName - should we use MessageNameResolver here?
```

Comment:
I don't think we have too...unless people expect that we are able to conjure up a `QualifiedName` for payloads that weren't Axon-payloads to begin with. I'd wager an application wouldn't register event payloads from other applications that didn't come from an Axon context, to be frank.

Hence, I think you can get rid of the TODO.

Reply:
Thanks, I thought the same :)